### DISCUSSION TITLE
How to make backup on TrueNAS

### DISCUSSION THREAD

**Discussion by DuMbh3aD** [photoprism/photoprism#3344] [General]
Hello,
How can I do a full photoprism backup?
I'm currently doing a Replication Task on the originals folder, but if something happens to the server how do I restore the database with meta data, people, albums, etc.?
I read the Advanced Backup Guide but I don't understand is where to write these commands, in system shell or in app shell or somewhere else?
If I Enable Custom Host Path for Photo Prism Storage Volume to do replication tasks on a storage volume will I break something?

![image](https://user-images.githubusercontent.com/124347033/232739709-dde592b4-98ac-4638-bb36-b928b2d04792.png)

**Comment by graciousgrey** [comment]
Unfortunately, we do not have a TrueNAS test device. Maybe another TrueNAS user or the TrueNAS team can help.

**Comment by DuMbh3aD** [comment]
I was referred to you by the TrueNAS community.
I hope someone knows how to make a backup

https://www.truenas.com/community/threads/full-backup-photoprism.108925/

**Reply by lastzero** [reply]
We don't maintain the TrueNAS integration, it's a third party project. If they let know how to run commands, you can use our docs. Or simply backup mariadb using any standard tool, so as documented by the vendor. I'm sure TrueNAS has a way to do that?

**Reply by DuMbh3aD** [reply]
I was able to make a backup with this command
`docker exec -t PhotoPrism sqlite3 /photoprism/storage/index.db .dump > photoprism-db.sql`
But I don't see an similar command in the documentation for Restoring Backups
If I make and save a zip file of the entire storage folder, will that be enough to restore later?

**Comment by erkk3** [comment]
Did you find a solution to this? I'm also trying to backup my photoprism from TrueNAS
